kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/DefaultConfiguration.java

```diff
@@ -66,5 +67,16 @@ public static Configuration get() {
    */
   public static void set(Configuration conf) {
     DefaultConfiguration.conf = conf;
+    DefaultConfiguration.initDone = true;
+  }
+
+  /**
+   * Initialize the default Hadoop {@link Configuration} if it has not been set.
+   */
+  public static void init(Configuration conf) {
+    if (!initDone) {
+      DefaultConfiguration.conf = conf;
+      DefaultConfiguration.initDone = true;
+    }
```

Comment:
All of the methods on this class should be `synchronized`. Otherwise this change looks good.